baremetal/machine_manager_test.go

```diff
@@ -0,0 +1,1697 @@
+					InfrastructureRef: *infrastructureRef,
+				},
+			},
+			Hosts:            []runtime.Object{&hostWithLabel},
+			BMMachine:        bmmconfig,
+			ExpectedHostName: hostWithLabel.Name,
+		},
+		"Choose the host with the right label": {
+			Machine: capi.Machine{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "machine1",
+					Namespace: "myns",
+				},
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "Machine",
+					APIVersion: capi.GroupVersion.String(),
+				},
+				Spec: capi.MachineSpec{
+					InfrastructureRef: *infrastructureRef2,
+				},
+			},
+			Hosts:            []runtime.Object{&hostWithLabel, &host2},
+			BMMachine:        bmmconfig2,
+			ExpectedHostName: hostWithLabel.Name,
+		},
+		"No host that matches required label": {
+			Machine: capi.Machine{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "machine1",
+					Namespace: "myns",
+				},
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "Machine",
+					APIVersion: capi.GroupVersion.String(),
+				},
+				Spec: capi.MachineSpec{
+					InfrastructureRef: *infrastructureRef3,
+				},
+			},
+			Hosts:            []runtime.Object{&host2, &hostWithLabel},
+			BMMachine:        bmmconfig3,
+			ExpectedHostName: "",
+		},
+		"Host that matches a matchExpression": {
+			Machine: capi.Machine{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "machine1",
+					Namespace: "myns",
+				},
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "Machine",
+					APIVersion: capi.GroupVersion.String(),
+				},
+				Spec: capi.MachineSpec{
+					InfrastructureRef: *infrastructureRef4,
+				},
+			},
+			Hosts:            []runtime.Object{&host2, &hostWithLabel},
+			BMMachine:        bmmconfig4,
+			ExpectedHostName: hostWithLabel.Name,
+		},
+		"No Host available that matches a matchExpression": {
+			Machine: capi.Machine{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "machine1",
+					Namespace: "myns-2",
+				},
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "Machine",
+					APIVersion: capi.GroupVersion.String(),
+				},
+				Spec: capi.MachineSpec{
+					InfrastructureRef: *infrastructureRef4,
+				},
+			},
+			Hosts:            []runtime.Object{&host2},
+			BMMachine:        bmmconfig4,
+			ExpectedHostName: "",
+		},
+		"No host chosen, invalid match expression": {
+			Machine: capi.Machine{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "machine1",
+					Namespace: "myns-3",
+				},
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "Machine",
+					APIVersion: capi.GroupVersion.String(),
+				},
+				Spec: capi.MachineSpec{
+					InfrastructureRef: *infrastructureRef5,
+				},
+			},
+			Hosts:            []runtime.Object{&host2, &hostWithLabel, &host1},
+			BMMachine:        bmmconfig5,
+			ExpectedHostName: "",
+		},
+	}
+
+	for name, tc := range testCases {
+		t.Logf("## TC-%s ##", name)
+		c := fakeclient.NewFakeClientWithScheme(scheme, tc.Hosts...)
+
+		actuator, err := newMachineManager(c, nil, nil, &tc.Machine, tc.BMMachine, klogr.New())
```

Comment:
Can you please replace all the "actuator" in this file by "machineMgr" or similar ? the term is from v1alpha1 and confusing now

Reply:
done